src/components/Timeline/updates/TimelineJourneyInstance.tsx

```diff
@@ -0,0 +1,105 @@
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import { Box, Button, Fade, Typography } from '@material-ui/core';
+import { FormattedMessage, useIntl } from 'react-intl';
+import React, { useEffect, useRef, useState } from 'react';
+
+import UpdateContainer from './elements/UpdateContainer';
+import ZetkinPersonLink from 'components/ZetkinPersonLink';
+import { ZetkinUpdateJourneyInstance } from 'types/updates';
+
+interface Props {
+  update: ZetkinUpdateJourneyInstance;
+}
+
+const TimelineJourneyInstance: React.FunctionComponent<Props> = ({
+  update,
+}) => {
+  const intl = useIntl();
+  const textRef = useRef<HTMLParagraphElement>();
+  const fieldToUpdate = Object.keys(update.details.changes)[0] as
+    | 'summary'
+    | 'title';
+  const [overflow, setOverflow] = useState<number>();
+  const [expandSummary, setExpandSummary] = useState<boolean>(false);
+  const [currentScroll, setCurrentScroll] = useState<number>(0);
+
+  useEffect(() => {
+    if (expandSummary && !!textRef) {
+      const offset = textRef?.current?.offsetTop as number;
+      window.scrollTo({
+        behavior: 'smooth',
+        top: offset - 100,
+      });
+      setTimeout(() => {
+        setCurrentScroll(offset - 100);
+        window.addEventListener('scroll', closeOnScroll);
+      }, 1000);
+    } else {
+      window.removeEventListener('scroll', closeOnScroll);
+    }
+  }, [expandSummary]);
+
+  useEffect(() => {
+    const text = textRef?.current;
+    if (text && text.scrollHeight > text.clientHeight) {
+      setOverflow(textRef.current?.scrollHeight);
+    }
+    return () => window.removeEventListener('scroll', closeOnScroll);
+  }, []);
```

Comment:
This is also really nice. It took me a while to understand what it did (maybe a comment would have been neat) but mostly I think that's because it's such a slick effect that I didn't realize as a user that it was even there.

Reply:
Added a comment